### PR TITLE
Add a DevCommand for enabling the minimap

### PR DESCRIPTION
**Pull request by penev92**
I don't know about you, but I really need my minimap, and I'm really tired of starting 20 games with an average length of 25 seconds when debugging something and having to deploy my MCV and build a radar each time.

### CODE REVIEW

File: OpenRA.Game/Traits/Player/DeveloperMode.cs [hunk 1]
```diff
@@ -38,6 +38,7 @@ public class DeveloperMode : IResolveOrder, ISync
 		[Sync] public bool PathDebug;
 		[Sync] public bool UnlimitedPower;
 		[Sync] public bool BuildAnywhere;
+		[Sync] public bool ProvideRadar;
```
Comment (pchote) [inline OpenRA.Game/Traits/Player/DeveloperMode.cs]: How about including this as part of DisableShroud?

File: OpenRA.Mods.Common/Commands/DevCommands.cs [hunk 2]
```diff
@@ -47,6 +47,7 @@ public void WorldLoaded(World w, WorldRenderer wr)
 			register("all", "toggles all cheats and gives you some cash for your trouble.");
 			register("crash", "crashes the game.");
 			register("levelup", "adds a specified number of levels to the selected actors.");
+			register("radar", "toggles the minimap if you don't have an actor that provides it.");
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Commands/DevCommands.cs]: We should be consistent wrt `radar` or `minimap`.
Also, like my toggleui PR, I'd like to ask for `toggle` to be in here somewhere as that is what you're actually doing.

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameRadarDisplayLogic.cs [hunk 3]
```diff
@@ -25,11 +26,12 @@ public IngameRadarDisplayLogic(Widget widget, World world)
 			var blockColor = Color.Transparent;
 			var radar = widget.Get<RadarWidget>("RADAR_MINIMAP");
 			radar.IsEnabled = () => radarEnabled;
+			var devMode = world.LocalPlayer.PlayerActor.Trait<DeveloperMode>();
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameRadarDisplayLogic.cs]: You are now requiring `DeveloperMode`. Either `Requires<>` it or do a null check.

Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameRadarDisplayLogic.cs]: This is widget code, not a trait.

File: OpenRA.Mods.Common/Commands/DevCommands.cs (hunk 2 above)
Comment (penev92) [inline OpenRA.Mods.Common/Commands/DevCommands.cs]: Well it does say "toggles the ..." :D

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Commands/DevCommands.cs]: That isn't part of the command.

**Comment by abcdefg30** [comment]
Can't we have an extra command for this/Why is it better to have this in the `DisableShroud` cheat?

**Comment by pchote** [comment]
It fits logically with the other actions of that cheat (DisableShroud is a poor name for what it actually does, which is to grant full visibility of the world), and I don't want to have to type a second console command to activate this separately from the vis.

**Comment by pchote** [comment]
:+1: code looks good and works ingame.

**Comment by obrakmann** [comment]
Looks good to me, and works as expected. :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/ee21130902b7ed6ee885f5089ed5298f5ecce525%5E...ee21130902b7ed6ee885f5089ed5298f5ecce525)

**Pull request merged**